syntax = "proto3";

package colbert;

service Server {
  rpc Search(Query) returns (QueryResult) {}
  rpc Pisa(Query) returns (QueryResult) {}
  rpc Serve(Query) returns (QueryResult) {}
  rpc DumpScores(Empty) returns (Empty) {}
}

message Query {
  string query = 1;
  int32 qid = 2;
  int32 k = 3;
}

message TopkResult {
  int32 pid = 1;
  int32 rank = 2;
  float score = 3;
}

message QueryResult {
  int32 qid = 1;
  repeated TopkResult topk = 2;
}

message Empty {
}
